autogluon_zeroshot/simulation/repository.py

```diff
@@ -21,18 +27,73 @@ def __init__(
             tabular_predictions: TabularModelPredictions,
             ground_truth: dict,
     ):
-
         self._tabular_predictions = tabular_predictions
         self._zeroshot_context = zeroshot_context
-        self._zeroshot_context.subset_datasets(self._tabular_predictions.datasets)
-        self._zeroshot_context.subset_models(self._tabular_predictions.list_models_available(present_in_all=True))
-        self._df_metadata = zeroshot_context.df_metadata if zeroshot_context.df_metadata is not None else pd.read_csv(Paths.data_root / "metadata" / "task_metadata.csv")
-        self._tid_to_name = dict(self._df_metadata[['tid', 'name']].values)
-        self._tid_to_name = {k: v for k, v in self._tid_to_name.items() if k in self._tabular_predictions.datasets}
-        self._name_to_tid = {v: k for k, v in self._tid_to_name.items()}
         self._ground_truth = ground_truth
         assert all(x in self._tid_to_name for x in self._tabular_predictions.datasets)
 
+    def print_info(self):
+        self._zeroshot_context.print_info()
+
+    @property
+    def _name_to_tid(self):
+        return self._zeroshot_context.dataset_to_tid_dict
+
+    @property
+    def _tid_to_name(self):
```

Comment:
I did not get the benefit of moving those to private properties compare to just initializing as standard python attributes.

Reply:
It ensures they remain correct if objects like `self._zeroshot_context` are modified